sway/desktop/render.c

```diff
@@ -169,10 +168,20 @@ static void render_surface_iterator(struct sway_output *output,
 		wlr_output->transform_matrix);
 
 	struct wlr_box dst_box = *_box;
-	struct wlr_box *clip_box = data->clip_box;
-	if (clip_box != NULL) {
-		dst_box.width = fmin(dst_box.width, clip_box->width);
-		dst_box.height = fmin(dst_box.height, clip_box->height);
+
+	// Only uses the view's width if it's the surfaces view (windows, not popups)
+	if (view && view->surface == surface) {
+		struct sway_container_state state = view->container->current;
+		dst_box.x = state.x - output->lx;
+		dst_box.y = state.y - output->ly;
+		dst_box.width = state.width;
+		dst_box.height = state.height;
+		if (state.border == B_PIXEL || state.border == B_NORMAL) {
+			dst_box.x += state.border_thickness;
+			dst_box.y += state.border_thickness;
+			dst_box.width -= state.border_thickness * 2;
+			dst_box.height -= state.border_thickness * 2;
+		}
 	}
```

Comment:
Fixed. I also removed the clip_box due to us using the view size instead. Would you prefer using the clipbox instead of this? (Moving this into `render_view_toplevels`)

Reply:
Using the clip_box actually makes the code a lot cleaner, so I'll go a head and push those changes